lib/jsdom/living/nodes/HTMLFrameElement-impl.js

```diff
@@ -54,7 +55,13 @@ function fetchFrame(serializedURL, frame, document, contentDoc) {
     contentDoc._encoding = encoding;
 
     const html = whatwgEncoding.decode(data, contentDoc._encoding);
-    contentDoc._htmlToDom.appendToDocument(html, contentDoc);
+
+    try {
+      parseDocument(html, contentDoc);
+    } catch (error) {
+      // Ignore parsing errors when XML is malformed
```

Comment:
I think we should fire a jsdomError, at least. The browser does a yellow screen of death; I'm not sure a "console error" is quite drastic enough, but it's a good start, probably.

Reply:
I updated the code to make it throw a jsdomError (https://github.com/jsdom/jsdom/pull/2522/files#diff-de5536e24fb5f7c11bd2a04cbecc0700R64).

After reading the spec, this line caught my attention (https://html.spec.whatwg.org/#read-xml):

> Error messages from the parse process (e.g., XML namespace well-formedness errors) may be reported inline by mutating the Document.

I am wondering if it makes more sense when the XML parser is parsing a document instead of throwing `DOMException` it should instead report the error directly in the document. This approach would remove the need to add a try/catch here and would probably stick closer to the spec. Any thoughts about this?

Reply:
Yes, replacing malformed documents with one that contains a `<parsererror>` element seems to be the approach taken by browsers. We already do so for documents created through DOMParser: https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/domparsing/DOMParser-impl.js#L20